### PR TITLE
*: upgrade etcd to master

### PR DESCRIPTION
**Pull request by nolouch**
Changes:
- Add [`embed.Config.InitialElectionTickAdvance`](https://github.com/coreos/etcd/pull/9591) to enable/disable initial election tick fast-forward.
   `embed.NewConfig()` would return `*embed.Config` with `InitialElectionTickAdvance` as true by 
default.
- Add `--pre-vote` flag to enable to run an additional Raft election phase
  will enbale later
- Define [`embed.CompactorModePeriodic`](https://godoc.org/github.com/coreos/etcd/embed#pkg-variables) for `compactor.ModePeriodic`.
  compatible
- Define [`embed.CompactorModeRevision`](https://godoc.org/github.com/coreos/etcd/embed#pkg-variables) for `compactor.ModeRevision`.
  compatible
- Change [`embed.Config.CorsInfo` in `*cors.CORSInfo` type to `embed.Config.CORS` in `map[string]struct{}` type](https://github.com/coreos/etcd/pull/9490).
  not used
- Remove [`embed.Config.SetupLogging`](https://github.com/coreos/etcd/pull/9572).Now logger is set up automatically based on [`embed.Config.Logger`, `embed.Config.LogOutputs`, `embed.Config.Debug` fields](https://github.com/coreos/etcd/pull/9572).
  will let test meet the panic problem, fixed.
- Add [`embed.Config.Logger`](https://github.com/coreos/etcd/pull/9518) to support [structured logger `zap`](https://github.com/uber-go/zap) in server-side.
  only use `zap` in test.
- Rename `embed.Config.SnapCount` field to [`embed.Config.SnapshotCount`](https://github.com/coreos/etcd/pull/9745), to be consistent with the flag name `etcd --snapshot-count`.
  not used
- Rename [**`embed.Config.LogOutput`** to **`embed.Config.LogOutputs`**](https://github.com/coreos/etcd/pull/9624) to support multiple log outputs.
  compatible
- Change [**`embed.Config.LogOutputs`** type from `string` to `[]string`](https://github.com/coreos/etcd/pull/9579) to support multiple log outputs.
  compatible
- Enable etcd server `raft.Config.CheckQuorum` when starting with `ForceNewCluster`.
  If new members are added to standalone node with`CheckQuorum == false` and the standalone node is the leader, this may reduce cluster availabilities, since the leader won't check quorum activities.

### CODE REVIEW

File: pkg/etcdutil/etcdutil_test.go [hunk 1]
```diff
@@ -42,6 +42,8 @@ func newTestSingleConfig() *embed.Config {
 	cfg.Name = "test_etcd"
 	cfg.Dir, _ = ioutil.TempDir("/tmp", "test_etcd")
 	cfg.WalDir = ""
+	cfg.Logger = "zap"
```
Comment (siddontang) [inline pkg/etcdutil/etcdutil_test.go]: do we use zap in production?

File: server/join.go [hunk 2]
```diff
@@ -73,8 +73,10 @@ func PrepareJoinCluster(cfg *Config) error {
 	}
 
 	// Cases with data directory.
+	// The path change from "member" to "member/wal"
+	// Reference: https://github.com/coreos/etcd/pull/9733
 	initialCluster := ""
-	if wal.Exist(path.Join(cfg.DataDir, "member")) {
+	if wal.Exist(path.Join(cfg.DataDir, "member/wal")) {
```
[the remark this answers is not in the document]
Comment (nolouch) [reply server/join.go]: yep. but better to keep the previous logic. address it.

File: pkg/etcdutil/etcdutil_test.go (hunk 1 above)
Comment (nolouch) [reply pkg/etcdutil/etcdutil_test.go]: etcd plan to use [`zap`](https://github.com/uber-go/zap) replace the `capnslog` logging. and we enable it in the test.

File: server/server.go [hunk 3]
```diff
@@ -285,6 +285,16 @@ func (s *Server) Run() error {
 	return nil
 }
 
+// SetUpTestMode for testing
+func (s *Server) SetUpTestMode() {
```
Comment (disksing) [inline server/server.go]: Can we do it in `server.CreateServer`? It looks verbose to call it in every tests...

**Comment by disksing** [comment]
LGTM. PTAL @siddontang @Connor1996

**Pull request merged**